opencog/learning/miner/miner-utils.scm

```diff
@@ -514,21 +517,29 @@
                (results (cog-fc miner-rbs source))
                ;; Fetch all relevant results
                (patterns (fetch-patterns texts-cpt minsup))
-               (patterns-lst (cog-outgoing-set patterns))
-
-               ;; Configure surprisingness backward chainer
-               ;; (I-Surprisingness for now)
-               (isurp-rbs (random-surprisingness-rbs-cpt))
-               (target (isurp-target texts-cpt))
-               (vardecl (isurp-vardecl))
-               (cfg-s (configure-isurp isurp-rbs surprisingness max-conjuncts))
-
-               ;; Run surprisingness in backward way
-               (isurp-results (cog-bc isurp-rbs target #:vardecl vardecl)))
-
-          (cog-pop-atomspace)
-
-          (desc-sort-by-tv-strength isurp-results)))))
+               (patterns-lst (cog-outgoing-set patterns)))
+
+          (if (equal? surprisingness 'none)
+
+              ;; No surprisingness, simple return the pattern list
+              (begin
+                (cog-set-atomspace! parent-as)
+                ;; TODO: delete tmp-as but without deleting its atoms
```

Comment:
ahh, OK, sorry.

Reply:
I think the root bug was that BIT was holding on to a temp atomspace even when parent atomspace is gone.this might be?? variant of that.